src/ng/directive/ngStyle.js

```diff
@@ -54,7 +54,11 @@
 var ngStyleDirective = ngDirective(function(scope, element, attr) {
   scope.$watchCollection(attr.ngStyle, function ngStyleWatchAction(newStyles, oldStyles) {
     if (oldStyles && (newStyles !== oldStyles)) {
-      forEach(oldStyles, function(val, style) { element.css(style, '');});
+      forEach(oldStyles, function(val, style) {
+        if (!(newStyles && newStyles[style])) {
+          element.css(style, '');
```

Comment:
Would it be possible to do `newStyles[style] = ''` instead (maybe also needing a `newStyles` null check + assign). This way we will only make a single `element.css(newStyles)` call at the end instead of multiple `.css` calls.

Also should this be an `undefined` check instead of just falsey check? What if `newStyles[style]` is `0`?

Reply:
For sure. Fixed in 95d2458289f42b9c399258cb0f4df038abc4dbb7.